### PR TITLE
Feature/superuser init refactor

### PR DESCRIPTION
**Pull request by mspector**
this PR makes a few important changes:

1. there is no longer any command-line interaction. All variables are loaded using a params.data file that gets passed into the script.

2. the code has been cleaned up so that it should now be much easier to add new templates. I'm working on adding an areachair console webfield template and a template for the webfield with tabs. 

3. superuser-init is no longer responsible for creating the admin user. I thought it was too clunky to include all of this together, so I plan to move it to a separate script.

### CODE REVIEW

File: admin/conference-template/params.data [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    e.g. "The 6th International Conference on Learning Representations"
+
+    LOCATION: A string describing the location of the conference.
+    e.g. "Vancouver Convention Center, Vancouver, BC, Canada"
+
+    EVENT_DATE: A string describing the date of the actual conference event.
+    e.g. "April 30 - May 3, 2018"
+
+    URL: A url for the conference website.
+    e.g. http://www.iclr.cc
```
Comment (pmandler) [inline admin/conference-template/params.data]: Should the example url be in quotes like the rest?

File: admin/conference-template/process/commentProcess.template [hunk 2]
```diff
@@ -21,15 +21,15 @@ function(){
       if(!selfComment){
         if(replytoNote.id == origNote.id){
           author_mail = {
-            "groups": origNote.content.authorids,
-            "subject": "Your submission to "+CONFERENCEPHRASE+" has received a comment",
-            "message": "Your submission to "+CONFERENCEPHRASE+" has received a comment.\n\nComment title: " + note.content.title + "\n\nComment: " + note.content.comment + "\n\nTo view the comment, click here: " + baseUrl+"/forum?id=" + note.forum +'&noteId='+note.id
+            'groups': origNote.content.authorids,
```
Comment (pmandler) [inline admin/conference-template/process/commentProcess.template]: Why the change from double quotes to single quotes?

Comment (mspector) [reply admin/conference-template/process/commentProcess.template]: I remember @zbialecki telling me that single quotes are preferred in javascript. is that right Zach?

Comment (zbialecki) [reply admin/conference-template/process/commentProcess.template]: This is correct, it's mainly a matter of style preference but we standardized on single quotes in the OpenReview codebase. Also object properties like group, subject and message do not need any quotes.

File: admin/conference-template/process/commentProcess.template [hunk 3]
```diff
@@ -21,15 +21,15 @@ function(){
...
           };
         } else {
           author_mail = {
-            "groups": replytoNote.signatures == '(anonymous)' ? [] : replytoNote.signatures,
-            "subject": "Your comment has received a response",
-            "message": "Your comment titled \""+replytoNote.content.title+"\" has received a response.\n\nComment title: " + note.content.title + "\n\nComment: " + note.content.comment + "\n\nTo view the comment, click here: " + baseUrl+"/forum?id=" + note.forum +'&noteId='+note.id
+            'groups': replytoNote.signatures == '(anonymous)' ? [] : replytoNote.signatures,
+            'subject': 'Your comment has received a response',
+            'message': 'Your comment titled \"'+replytoNote.content.title+'\" has received a response.\n\nComment title: ' + note.content.title + '\n\nComment: ' + note.content.comment + '\n\nTo view the comment, click here: ' + baseUrl+'/forum?id=' + note.forum +'&noteId='+note.id
```
Comment (zbialecki) [inline admin/conference-template/process/commentProcess.template]: You don't need to escape the double quote if you used single quotes at the beginning of the string.

File: admin/superuser-init.py [hunk 4]
```diff
@@ -1,290 +1,139 @@
...
+    ext = '.py' if 'python' in template_path else '.js'
+    newfile_path = directory_path + '/' + template_path.replace('.template', ext)
+
+    with open(os.path.join(os.path.dirname(__file__), './conference-template/{0}'.format(template_path))) as template:
+        template_string = template.read()
+
+    for replacement in data:
+        template_string = template_string.replace(replacement, data[replacement])
+
+    if not os.path.exists(newfile_path) or overwrite:
```
Comment (pmandler) [inline admin/superuser-init.py]: Move this check up to line 57. No reason to recreate the file if it's not going to be used.

File: admin/conference-template/params.data [hunk 5]
```diff
@@ -0,0 +1,82 @@
+{
```
Comment (melisabok) [inline admin/conference-template/params.data]: Why this is not called params.json?

File: admin/conference-template/params.data [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+
+    URL: A url for the conference website.
+    e.g. "http://www.iclr.cc"
+
+    DEADLINE: A string describing the deadline of the conference.
+    e.g. "Submission Deadline: 5:00PM Eastern Standard Time, October 27, 2017"
+
+    */
+
+    "<<TITLE>>": "TITLE",
```
Comment (melisabok) [inline admin/conference-template/params.data]: I would remove '<<' and '>>' from the keys and add them when we build the templates with the configuration. It is more descriptive to have the name only.

File: admin/README.md [hunk 7]
```diff
@@ -1,48 +1,14 @@
 # Admin workflow
-To kick off a conference, the OpenReview superuser needs to:
+To kick off a conference, the OpenReview superuser should:
 - Create a conference directory under openreview-scripts/venues
-- Create and post the conference's root group to the system
-- Create the conference administrator account (Optional)
+- Copy the file openreview-scripts/admin/conference-template/params.data and fill it in with conference-specific variables
```
Comment (melisabok) [inline admin/README.md]: copy and paste it in which place? is there a default location for the params.data?

File: admin/README.md [hunk 8]
```diff
@@ -1,48 +1,14 @@
...
-```
-
-`superuser-init.py` also generates a file at the top level of the conference directory called `params.json`. This file contains all the parameters that were initially passed into the script's text prompts. Passing the path of `params.json` into the `--data` flag allows the user to skip the text prompts and generate new files using those parameters. When combined with the `--overwrite` flag, this is useful for making changes to the conference configuration.
-
-```
 Example:
 
->> superuser-init.py --conf myconf.org/MYCONF/2017 --data ./params.json 
+```
+>> superuser-init.py --conf myconf.org/MYCONF/2017 --data path/to/params.data
```
Comment (melisabok) [inline admin/README.md]: superuser-init could read the params.data from a default location.

Comment (melisabok) [reply admin/README.md]: what is the conf param? the folder of the conference? 'conf' is confusing with the 'configuration' word. Could we use --venue-path or --venue ?

File: admin/superuser-init.py [hunk 9]
```diff
@@ -1,290 +1,139 @@
...
 
-Gathers information to fill in config.py and the webfield/conf.html.
-Creates directories and groups for the entire CONF directory structure.
-Creates sub-directores python/, process/, data/ and webfield/.
-Creates config.py, admin-init.py, submissionProcess.js, commentProcess.js and conf.html.
-Optionally creates admin user profile
+
 """
 
+def parse_json(file):
```
Comment (melisabok) [inline admin/superuser-init.py]: Why don't use this to load a json?

```
import json

with open('strings.json') as json_data:
    d = json.load(json_data)
    print(d)
```

File: admin/superuser-init.py [hunk 10]
```diff
@@ -1,290 +1,139 @@
...
-    client.add_members_to_group(admin_group, [username])
-    print "Added %s to %s" % (username, args.conf)
+directory_path = os.path.join(os.path.dirname(__file__), '../venues/{0}'.format(args.conf))
+conference_group_id = args.conf
+
+# load data
+data = parse_json(args.data)
+
+# build the directory structure
+build_directory(directory_path)
```
Comment (melisabok) [inline admin/superuser-init.py]: you can send the list of folders as a parameter, the same that you do with templates.

File: admin/superuser-init.py [hunk 11]
```diff
@@ -1,290 +1,139 @@
...
+# generate new files from templates
+for file in templates:
+    generate_file(file, directory_path, data, overwrite = args.overwrite)
+
+groups = build_groups(conference_group_id)
+
+print "created the following groups:"
+for g in groups: print g
+
+post_groups = raw_input("Post groups to {0}? (y/[n]): ".format(client.baseurl)).lower()
```
Comment (melisabok) [inline admin/superuser-init.py]: why did you add this question?

File: admin/conference-template/params.data (hunk 5 above)
Comment (mspector) [reply admin/conference-template/params.data]: I call it .data because it's not really a proper .json file. Json doesn't support comments, but I want there to be comments in the file so that it can be easily read and edited by a person.

This is also the reason I wrote a special parse_json function. the native json parser throws an error if there are comments in the file.

File: admin/README.md (hunk 7 above)
Comment (mspector) [reply admin/README.md]: I guess I'm imagining that users will create a directory based on the group's ID, but it should work even if it's in a different place

e.g. they'll create the directory venues/TestConf.org/2018/Conference, and then put the params file there.

I'll update the readme to say that

File: admin/README.md (hunk 8 above)
Comment (mspector) [reply admin/README.md]: ok, I will make the --data parameter optional and look for it in the venue directory

File: admin/superuser-init.py (hunk 9 above)
Comment (mspector) [reply admin/superuser-init.py]: I said this above but the reason for this is that the json format doesn't accept comments, and I want comments in the params file

File: admin/conference-template/params.data (hunk 5 above)
Comment (melisabok) [reply admin/conference-template/params.data]: Ok, so it is not a json file. What about to call it .properties. This is more standard and I guess python has a good way to load it. 

```
CONFERENCE_NAME=This is a the conference name
```

File: admin/superuser-init.py (hunk 11 above)
Comment (mspector) [reply admin/superuser-init.py]: so that the user is reminded of which baseurl the client is connected to

do you think it's not needed?

Comment (melisabok) [reply admin/superuser-init.py]: I don't know, how is going to run this script, @pmandler ? 

and what happens if you say no? the conference files are created but not the groups? it is weird.

Comment (mspector) [reply admin/superuser-init.py]: ok, fair enough. i'll remove it

**Comment by mspector** [comment]
After examining the program chair webfield, I'm realizing that it's not really ready to be added into the templates yet. I'm working on a different branch with that webfield and the area chair webfield.

It turns out that both of these webfields are pretty specific. It will take some thinking to figure out the best way to make them general

File: admin/README.md [hunk 12]
```diff
@@ -1,48 +1,14 @@
...
```
Comment (melisabok) [inline admin/README.md]: I guess you need to change the doc here

**Comment by melisabok** [comment]
Sounds good, this was a very nice refactor!

**Comment by mspector** [comment]
I think it's ready, but it's not urgently needed
